src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -613,10 +595,8 @@ private Compilation CreateEmptyCompilation()
                     return await FinalizeCompilationAsync(
                         solution,
                         compilationWithoutGenerators,
-                        authoritativeGeneratedDocuments: null,
-                        nonAuthoritativeGeneratedDocuments: state.GeneratorInfo.Documents,
+                        state.GeneratorInfo.WithDriver(generatorDriver),
```

Comment:
Previously `authoritativeGeneratedDocuments` was null'd, do we need to explicitly set `DocumentsAreFinal` to false here?

Reply:
When the InProgressState was created, the constructor explicitly calls WithDocumentsAreFinal(false).